### DISCUSSION TITLE
skin elements do not get scaled properly when opening skin editor

### DISCUSSION THREAD

**Discussion by mhdimo** [ppy/osu#18684] [Q&A]
### Type

Game behaviour

### Bug description

skin elements do not get scaled properly when opening skin editor, causing much confusion where to position them, especially with skins that instead of the overlay of the hp bar uses a overlay that covers the whole display to focus more in game only 

### Screenshots or videos

![osu_2022-06-14_11-06-15-1](https://user-images.githubusercontent.com/62386208/173540087-48ca87a2-18b1-4531-bab0-f8a693e4f9e9.jpg)

### Version

2022.612.0

### Logs

[performance.log](https://github.com/ppy/osu/files/8898292/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8898293/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8898294/updater.log)
[network.log](https://github.com/ppy/osu/files/8898295/network.log)

**Comment by peppy** [comment]
Using such elements in lazer are not supported by the skin editor. Please avoid using them in it.

On top of that, I don't get what you mean about them being mispositioned. You're going to have to explain better.

**Reply by integerrr** [reply]
The "mispositioned" thing is what I tried to bring up in #14887:
> there are UI elements (hp bars with a playfield border, accuracy counter that pads space to centralise the numbers etc.) that will look funky when gameplay aspect ratio changes entering the skin editor i.e. what you have in the editor wont match what you will see once exiting skin editor.

Basically nothing is really mispositioned, but the hp element in op's case (being more than just hp) gets moved around on its anchor point and doesn't necessarily match the position seen in the editor, depending on where the anchor is. 

I've avoided this problem by using simpler UI elements i.e. HP overlay being just the HP bar etc.

**Reply by peppy** [reply]
Sure, but the default position of the HP bar is top left anchor, which should not move in an unexpected way. So I'm not sure exactly what the user is experiencing here.

**Reply by Leochan6** [reply]
I am facing a similar issue where the skin editor does not position components correctly and once saved the actual positioning of components shifts. I believe this might have something to do with the aspect ratio of the editor window.

The left component selector bar and right component settings bar take up space while there are no top and bottom bars to maintain the aspect ratio. This causes the anchor points the sides (like top left of bottom right) to shift positions on the screen.

A potential solution is to use center anchor and origin points, but this makes it harder to edge align components.

**Reply by peppy** [reply]
Can you make a video of what you're trying to do? If things are moving, it's almost guaranteed the anchor or origin are set wrong.

**Reply by Leochan6** [reply]
Attached is a video where I create a big black box at the center of the screen, but since the editor is not the same aspect ratio of the monitor, it doesn't match the edges.

https://user-images.githubusercontent.com/9027044/194469035-9e117184-227d-4c8a-b761-1b903c90c80c.mp4

**Reply by peppy** [reply]
What you are trying to do is scale an element using relative size. The editor doesn't support relative elements currently. I'm not sure whether it will in the future, but it may be feasible.

**Reply by Leochan6** [reply]
I do not need to scale an element relative to the screen size. All that needs to happen is for the components to be in the same position in edit mode and preview mode. Adjusting the editor to have the same aspect ratio and resolution of actual gameplay may fix this. I understand this means different versions of skins will have to be made to support different aspect ratios and resolutions.

A workaround I found is to resize the component to the right aspect ratio, move the a significant portion off to the side screen, resize to the correct height, then moving it back to the center. This allows it to fill the full screen.

**Reply by peppy** [reply]
I think you might be missing the point a touch. You can make skins work for all aspect ratios by using anchor points. That is how we've made the *whole game* work as such. It's using the exact same specifications.

You can guarantee screen positioning relative to an edge or center.
